assets/js/log-viewer/store.ts

```diff
@@ -17,16 +17,37 @@ interface LogStore {
   formattedLogLines: string;
   addLogLines: (newLogs: LogLine[]) => void;
   highlightedRanges: { start: number; end: number }[];
+  desiredLogLevels: string[] | undefined;
+  setDesiredLogLevels: (desiredLogLevel: string[] | undefined) => void;
 }
 
-function findSelectedRanges(logs: LogLine[], stepId: string | undefined) {
+// check if a log matches the desired log level
+function matchesLogFilter(
+  log: LogLine,
+  desiredLogLevels: string[] | undefined
+): boolean {
+  if (!desiredLogLevels) return true;
+
+  return desiredLogLevels.includes(log.level);
+}
+
+function findSelectedRanges(
+  logs: LogLine[],
+  stepId: string | undefined,
+  desiredLogLevels: string[] | undefined
+) {
   if (!stepId) return [];
 
   const { ranges } = logs.reduce<{
     ranges: { start: number; end: number }[];
     marker: number;
   }>(
     ({ ranges, marker }, log) => {
+      // Skip logs that don't match the desired log level
+      if (!matchesLogFilter(log, desiredLogLevels)) {
```

Comment:
For a `findSelectedRanges` I would expect the matching logs to be filtered. If it works down the line, maybe rename to `rejectSelectedRanges`?

Reply:
What makes it harder to understand is that it it's trying to find ranges and does operations that sounds the opposite like skipping logs when doesn't match `desiredLogLevels` (as if `findSelected` is conflicting to _not_ matching the desired logs).

I understand the intention was super, to abbreviate the reduce, it's more about the names.

Same here on this same function, for findSelected passing a stepId and negating it with:
```
if (log.step_id !== stepId) {
        return { ranges, marker: nextMarker };
      }
```